pkgs/development/mobile/androidenv/mkrepo.rb

```diff
@@ -154,7 +154,7 @@ def normalize_license license
   license = license.dup
   license.gsub!(/([^\n])\n([^\n])/m, '\1 \2')
   license.gsub!(/ +/, ' ')
-  license
+  license.strip
```

Comment:
I think you actually want:

```ruby
license.strip!
license
```

So the strip is performed in place.

Reply:
Not really! Apparently `sdkmanager` strips "mips-android-sysimage-license's text\n" to "mips-android-sysimage-license's text", then make a hash. I don't want the `nil`! So `strip` is correct. Check

- https://ruby-doc.org/3.2.0/String.html#method-i-strip
- https://www.ruby-docs.org/v/3.1.2/String/instance_method-strip!

Reply:
By returning license at the end after the strip!, you'll never return nil

Reply:
I try to understand your point by running it in IRB, but without success! would you elaborate on the corner case that would go wrong with `strip`? and how `strip!` would solve it? Thanks!

Reply:
I reviewed it again, and the only thing that can be the difference is the performance, because `strip` makes a copy, but `strip!` updates the `self`, so let me apply the changes.

Reply:
check out [7962f8d](https://github.com/NixOS/nixpkgs/pull/208137/commits/7962f8d6acad5d578e1b05903a27b15a6ef9fc97)